openvdb/points/PointMask.h

```diff
@@ -0,0 +1,384 @@
+///////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2012-2017 DreamWorks Animation LLC
+//
+// All rights reserved. This software is distributed under the
+// Mozilla Public License 2.0 ( http://www.mozilla.org/MPL/2.0/ )
+//
+// Redistributions of source code must retain the above copyright
+// and license notice and the following restrictions and disclaimer.
+//
+// *     Neither the name of DreamWorks Animation nor the names of
+// its contributors may be used to endorse or promote products derived
+// from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
+// LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
+//
+///////////////////////////////////////////////////////////////////////////
+
+/// @file points/PointMask.h
+///
+/// @author Dan Bailey
+///
+/// @brief  Methods for extracting masks from VDB Point grids.
+
+#ifndef OPENVDB_POINTS_POINT_MASK_HAS_BEEN_INCLUDED
+#define OPENVDB_POINTS_POINT_MASK_HAS_BEEN_INCLUDED
+
+#include <openvdb/openvdb.h>
+#include <openvdb/tools/Prune.h> // pruneInactive
+#include <openvdb/tools/ValueTransformer.h> // valxform::SumOp
+
+#include "PointDataGrid.h"
+#include "IndexFilter.h"
+
+#include <tbb/combinable.h>
+
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace points {
+
+
+/// @brief Extract a Mask Grid from a Point Data Grid
+/// @param grid             the PointDataGrid to extract the mask from.
+/// @param includeGroups    a vector of VDB Points groups to be included (default is all).
+/// @param excludeGroups    a vector of VDB Points groups to be excluded (default is none).
+/// @note this method is only available for Bool Grids and Mask Grids
+template <typename PointDataGridT,
+          typename MaskT = typename PointDataGridT::template ValueConverter<bool>::Type>
+inline typename std::enable_if<std::is_same<typename MaskT::ValueType, bool>::value,
+    typename MaskT::Ptr>::type
```

Comment:
how about MaskGrid support?

Reply:
MaskT::ValueType evaluates to bool for both BoolGrids and MaskGrids.